### ISSUE TITLE
Uncaught Shiny error breaks RStudio's "Run in Window" feature

### ISSUE THREAD

**Issue by cpsievert** [rstudio/rstudio#12569]
Initially reported in https://github.com/rstudio/shiny/issues/3761

### System details

    RStudio Edition :    Desktop
    RStudio Version :   2022.12.0+353
    OS Version      :      Mac OS 13.1
    R Version       :        4.2.2

### Steps to reproduce the problem

Save the following Shiny app to disk. Then click the "Run App" button with "Run in Window" checked. When the app opens, click the "Boom" button. After the app crashes, click "Run App" again (to run the app again RStudio's Window browser). This time, the app's UI doesn't appear (and the JS console gives an error of `desktop_allow_navigation: Error invoking remote method 'desktop_allow_navigation': TypeError: Object has been destroyed`).

```r
library(shiny)

ui <- fluidPage(
  actionButton("boom", "Boom")
)

server <- function(input, output) {
  observeEvent(input$boom, {
    stop("boom")
  })
}

shinyApp(ui, server)

```  

### Describe the behavior you expected

I'd expect to be able to repeatedly run the Shiny app in RStudio's window (and click the "Boom" button to generate an uncaught error)

**Comment by sharon-wang** [comment]
This hasn't been triaged, but seems to have the same root issue as #12468. PR https://github.com/rstudio/rstudio/pull/12686 should resolve this bug.

**Comment by sharon-wang** [comment]
Here's a demo of the error reproduction and fix from #12686. `control+C` is used to exit the R session before re-running the app.

### Reproduction (https://dailies.rstudio.com/version/2023.03.0-daily+293/)
https://user-images.githubusercontent.com/25834218/218549918-578cb714-96b2-4b8c-a2e6-92b62d14fc4a.mov

### Behaviour after Fix from #12686
https://user-images.githubusercontent.com/25834218/218550175-94bc8f63-1f2b-4566-b8cc-16f05cf58433.mov

**Comment by jonvanausdeln** [comment]
Tested on 2023.03.0-daily+316  Desktop
Windows11 and MacOS 12.4

As described, I'm able to run the app after initial run, but only after interrupting the R session (via ctrl-c or stop button). Not sure of the uncaught error should interrupt the R session by itself.

**Comment by jonvanausdeln** [comment]
After some thought and discussion, I'm closing this as the initial repro is fixed.  If the viewer closing after the error is a problem, that can be opened in another issue.
